source/viewmodel.ios.ts

```diff
@@ -365,7 +431,7 @@ class AssetPH extends Asset {
         return undefined;
     }
 
-    data(): Thenable<any> {
+    data(): Promise<any> {
```

Comment:
@PanayotCankov The reason I chose to stick with the Promise class is because the Thenable interface does not provide the `catch` method, which could cause some issues when trying to consume errors for TS users. They would have to cast the result so they could call `catch` with their handler. For JS users, the return type doesn't matter.